File: 5KYU/Calculating with Functions/solution.js
const zero = (op) => op === undefined ? 0 : op(0);
const one = (op) => op === undefined ? 1 : op(1);
const two = (op) => op === undefined ? 2 : op(2);
const three = (op) => op === undefined ? 3 : op(3);
const four = (op) => op === undefined ? 4 : op(4);
const five = (op) => op === undefined ? 5 : op(5);
const six = (op) => op === undefined ? 6 : op(6);
const seven = (op) => op === undefined ? 7 : op(7);
const eight = (op) => op === undefined ? 8 : op(8);
const nine = (op) => op === undefined ? 9 : op(9);

const plus = (b) => (a) => a + b;
const minus = (b) => (a) => a - b;
const times = (b) => (a) => a * b;
const dividedBy = (b) => (a) => Math.floor(a / b);

File: 5KYU/Calculating with Functions/solution.rb
def zero(op = nil)
  op.nil? ? 0 : op.call(0)
end

def one(op = nil)
  op.nil? ? 1 : op.call(1)
end

def two(op = nil)
  op.nil? ? 2 : op.call(2)
end

def three(op = nil)
  op.nil? ? 3 : op.call(3)
end

def four(op = nil)
  op.nil? ? 4 : op.call(4)
end

def five(op = nil)
  op.nil? ? 5 : op.call(5)
end

def six(op = nil)
  op.nil? ? 6 : op.call(6)
end

def seven(op = nil)
  op.nil? ? 7 : op.call(7)
end

def eight(op = nil)
  op.nil? ? 8 : op.call(8)
end

def nine(op = nil)
  op.nil? ? 9 : op.call(9)
end

def plus(b)
  lambda {|a| a + b}
end

def minus(b)
  lambda {|a| a - b}
end

def times(b)
  lambda {|a| a * b}
end

def divided_by(b)
  lambda {|a| a / b}
end